### ISSUE TITLE
Chart view forgets x and y axis at every update

### ISSUE THREAD

**Issue by garycmartin** [kengruven/strukt-bugs#19]
When interacting live with the chart view, the two fields selected for use as the x and y axis are reset to the first field each time the pipeline is edited or re-run. It would be great if strukt remembered the chosen fields, or had some way of setting them via a pipeline operation, so you can see live chart updates without having to manually use the UI dropdown menus to reselect the axis fields each time.

**Comment by kengruven** [comment]
Yes, this is a problem that affects all field popups (text, map, etc.).

(Technical nonsense: The basic way it works is that it accepts streaming input, and has no memory of its own.  When it first sees a value of the correct type, it enables that item.  If it's the first enabled item, it selects it, too.  When a pipeline runs again, it momentarily has no data, so the popup gets reset.)

I seem to recall there was some reason I thought I couldn't just have it remember what field the user most recently selected, but at the moment I'm not sure what it was.

I'll take another swing at this.

**Comment by kengruven** [comment]
Strukt 1.4 now remembers the last field you've chosen (for charts, text, HTML, and maps), so if you re-run the pipeline (e.g., by editing it), it will show that same field again.

**Comment by garycmartin** [comment]
Just tested and is working well, thanks! Only extra suggestion would be to remember the fields between view changes as well (I often find myself toggling between Table and Chart view while tinkering with a data set).

**Comment by kengruven** [comment]
Yeah, that's a problem.  All the other user-interface state associated with each output view (e.g., the map zoom, or the table selection and scroll position) isn't remembered between view changes, either, and it should be.

I'll take a look at this...
